File: src/main/java/com/example/eventsphere/booking/BookingRequestDTO.java
package com.example.eventsphere.booking;


import com.example.eventsphere.attendee.AttendeeDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;


@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class BookingRequestDTO {

    private Long eventId;
    private List<AttendeeDTO> attendees;
}

File: lib/models/booking_request_dto.dart
import 'package:event_sphere/models/attendee_form_model.dart';

class BookingRequestDTO{
  final int eventId;
  final List<AttendeeFormModel> attendees;

  BookingRequestDTO({
    required this.eventId,
    required this.attendees
  });


  Map<String,dynamic> toJson()
  {
    return {
      'eventId':eventId,
      'attendees':attendees.map((attendee)=>attendee.toJson()).toList()
    };
  }
}